### PR TITLE
Enhance resourceprocessor

### PR DESCRIPTION
**Pull request by omrozowicz-splunk**
N/A

### CODE REVIEW

File: processor/resourcedetectionprocessor/config.go [hunk 1]
```diff
@@ -61,13 +96,43 @@ type DetectorConfig struct {
 func (d *DetectorConfig) GetConfigFromType(detectorType internal.DetectorType) internal.DetectorConfig {
 	switch detectorType {
 	case ec2.TypeStr:
-		return d.EC2Config
+		ec2Config := ec2.CreateDefaultConfig()
+		return ec2Config
+	case ecs.TypeStr:
+		ecsConfig := ecs.CreateDefaultConfig()
+		return ecsConfig
```
Comment (dmitryax) [inline processor/resourcedetectionprocessor/config.go]: Just for brevity :)
```suggestion
		return ecs.CreateDefaultConfig()
```

File: processor/resourcedetectionprocessor/factory.go [hunk 2]
```diff
@@ -97,7 +97,6 @@ func createDefaultConfig() component.Config {
 		Detectors:          []string{env.TypeStr},
 		HTTPClientSettings: defaultHTTPClientSettings(),
 		Override:           true,
-		Attributes:         nil,
```
Comment (dmitryax) [inline processor/resourcedetectionprocessor/factory.go:89]: We should not drop this config right away. It should be supported for some time as a deprecated option. See https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/splunkhecexporter/config.go#LL88C2-L88C16 as an example

File: processor/resourcedetectionprocessor/internal/aws/ec2/ec2.go [hunk 3]
```diff
@@ -17,17 +17,18 @@ package ec2 // import "github.com/open-telemetry/opentelemetry-collector-contrib
...
 	"github.com/aws/aws-sdk-go/service/ec2"
 	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor/internal/aws/ec2/internal/metadata"
 	"go.opentelemetry.io/collector/pdata/pcommon"
 	"go.opentelemetry.io/collector/processor"
 	conventions "go.opentelemetry.io/collector/semconv/v1.6.1"
 	"go.uber.org/zap"
+	"net/http"
+	"reflect"
+	"regexp"
```
Comment (dmitryax) [inline processor/resourcedetectionprocessor/internal/aws/ec2/ec2.go]: These must be in the first group. Linter will complain :)

File: processor/resourcedetectionprocessor/internal/aws/ec2/ec2.go [hunk 4]
```diff
@@ -57,19 +59,38 @@ func NewDetector(set processor.CreateSettings, dcfg internal.DetectorConfig) (in
...
+		}
+	}
+}
+
 func (d *Detector) Detect(ctx context.Context) (resource pcommon.Resource, schemaURL string, err error) {
 	res := pcommon.NewResource()
-	if _, err = d.metadataProvider.InstanceID(ctx); err != nil {
-		d.logger.Debug("EC2 metadata unavailable", zap.Error(err))
-		return res, "", nil
-	}
```
Comment (dmitryax) [inline processor/resourcedetectionprocessor/internal/aws/ec2/ec2.go]: why is this removed?

File: processor/resourcedetectionprocessor/internal/aws/ec2/ec2.go [hunk 5]
```diff
@@ -81,16 +102,22 @@ func (d *Detector) Detect(ctx context.Context) (resource pcommon.Resource, schem
...
-	attr.PutStr(conventions.AttributeCloudPlatform, conventions.AttributeCloudPlatformAWSEC2)
-	attr.PutStr(conventions.AttributeCloudRegion, meta.Region)
-	attr.PutStr(conventions.AttributeCloudAccountID, meta.AccountID)
-	attr.PutStr(conventions.AttributeCloudAvailabilityZone, meta.AvailabilityZone)
-	attr.PutStr(conventions.AttributeHostID, meta.InstanceID)
-	attr.PutStr(conventions.AttributeHostImageID, meta.ImageID)
-	attr.PutStr(conventions.AttributeHostType, meta.InstanceType)
-	attr.PutStr(conventions.AttributeHostName, hostname)
+
+	setResourceAttributes(dictionary, resourceAttributes, attr)
```
Comment (dmitryax) [inline processor/resourcedetectionprocessor/internal/aws/ec2/ec2.go]: I assume you added this function for easier future maintenance, but `setResourceAttributes` is hard to read, and we might generate helpers wrapping the _if enabled_ check in the future as it's done for metrics scraping receivers. So I'd suggest keeping it simple for now and writing a bit of boilerplate instead:
```
if resourceAttributes.CloudAccountID.Enabled {
	attr.PutStr(conventions.AttributeCloudAccountID, meta.AccountID)
}
```

File: processor/resourcedetectionprocessor/config.go (hunk 1 above)
Comment (omrozowicz-splunk) [reply processor/resourcedetectionprocessor/config.go]: Done

File: processor/resourcedetectionprocessor/internal/aws/ec2/ec2.go (hunk 3 above)
Comment (omrozowicz-splunk) [reply processor/resourcedetectionprocessor/internal/aws/ec2/ec2.go]: Done

File: processor/resourcedetectionprocessor/internal/aws/ec2/ec2.go (hunk 4 above)
Comment (omrozowicz-splunk) [reply processor/resourcedetectionprocessor/internal/aws/ec2/ec2.go]: By mistake I think, brought it back

File: processor/resourcedetectionprocessor/internal/aws/ec2/ec2.go (hunk 5 above)
Comment (omrozowicz-splunk) [reply processor/resourcedetectionprocessor/internal/aws/ec2/ec2.go]: Thanks! I put this method everywhere, it is really much simpler, as sometimes setting these attributes required some further checks and everything was getting so complicated...

**Review by mx-psi — COMMENTED** [review]
The approach makes sense to me. I think if you run `golangci-lint run --fix` it will fix all the import groups
